# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/Shader.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/Shader.h
--- a/gfx/angle/checkout/src/libANGLE/Shader.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/Shader.h	Fri Nov 08 17:06:00 2019 +0000
@@ -31,6 +31,7 @@
 class GLImplFactory;
 class ShaderImpl;
 class ShaderSh;
+class WaitableCompileEvent;
 }  // namespace rx
 
 namespace angle
@@ -189,6 +190,8 @@
     const std::string &getCompilerResourcesString() const;
 
   private:
+    struct CompilingState;
+
     ~Shader() override;
     static void GetSourceImpl(const std::string &source,
                               GLsizei bufSize,
@@ -208,10 +211,7 @@
 
     // We keep a reference to the translator in order to defer compiles while preserving settings.
     BindingPointer<Compiler> mBoundCompiler;
-    ShCompilerInstance mShCompilerInstance;
-    std::shared_ptr<CompileTask> mCompileTask;
-    std::shared_ptr<angle::WorkerThreadPool> mWorkerPool;
-    std::shared_ptr<angle::WaitableEvent> mCompileEvent;
+    std::unique_ptr<CompilingState> mCompilingState;
     std::string mCompilerResourcesString;
 
     ShaderProgramManager *mResourceManager;